bitcoin/src/address/mod.rs

```diff
@@ -450,6 +460,16 @@ impl<V: NetworkValidation> Address<V> {
 
     /// Marks the network of this address as unchecked.
     pub fn into_unchecked(self) -> Address<NetworkUnchecked> { Address(self.0, PhantomData) }
+
+    /// Returns the [`NetworkKind`] of this address.
+    pub fn network_kind(&self) -> NetworkKind {
+        use AddressInner::*;
+        match self.0 {
+            P2pkh { hash: _, ref network } => *network,
+            P2sh { hash: _, ref network } => *network,
+            Segwit { program: _, ref hrp } => NetworkKind::from(*hrp),
+        }
+    }
```

Comment:
Might be worth also having `is_mainnet` method (and possibly also `is_a_test`).

Reply:
I'm leaning toward not because these methods are one-liners (once this method is in) and it's extra API surface.

Reply:
Well, these are common in Rust libraries. But maybe it's enough to have them on `NetworkKind` so once can call `network_kind().is_mainnet()`.